packages/dialog/index.tsx

```diff
@@ -0,0 +1,322 @@
+// The MIT License
+//
+// Copyright (c) 2019 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as React from 'react';
+import classnames from 'classnames';
+
+// @ts-ignore no .d.ts file
+import {MDCDialogFoundation, MDCDialogAdapter, util} from '@material/dialog/dist/mdc.dialog';
+// @ts-ignore no .d.ts file
+import {ponyfill} from '@material/dom/dist/mdc.dom';
+/* eslint-disable no-unused-vars */
+import DialogContent, {DialogContentProps} from './DialogContent';
+import DialogFooter, {DialogFooterProps} from './DialogFooter';
+import DialogTitle, {DialogTitleProps} from './DialogTitle';
+/* eslint-enable no-unused-vars */
+import DialogButton from './DialogButton';
+import {cssClasses, LAYOUT_EVENTS} from './constants';
+import {FocusTrap} from 'focus-trap';
+
+export type ChildTypes<ChildProps> = DialogTitle<ChildProps> | DialogContent<ChildProps>| DialogFooter<ChildProps>;
+export type ChildProps<T> = DialogTitleProps<T> | DialogContentProps<T> | DialogFooterProps<T>;
+
+export interface DialogProps<T> extends React.HTMLProps<T> {
```

Comment:
`T` here is representing 4 different elements (title/content/footer/dialog). You need to break T into 4 different arguments. Or you could force the user to pass in DialogTitleProps/ContentProps/FooterProps as the 3 other arguments. Does that make sense? If you have or find a better solution please try it out.

Reply:
This one too still needs to be addresed